File: potd4/fib.java
import java.io.*;
import java.util.*; 

class fib {
	
	public static void main (String [] args) throws IOException {
		
		Scanner scan = new Scanner(System.in); 
		
		int N = scan.nextInt(); 
		int temp1 = 1, temp2 = 1, now = 0; 
		
		for(int i =  0; i < N ; i++){
				now = (temp1 + temp2)% 2017; 
				temp1 = temp2; 
				temp2 = now; 
		}
		
		System.out.println(now);
		
		System.exit(0);
	}
}

File: potd4/fib.cc
#include <iostream>

using namespace std; 

int main(){
	
	int N; 
	cin>> N; 
	
	int t1 = 1, t2 = 1, now = t1 +t2; 
	
	for(int i = 0 ; i < N ; i++){
		now = (t1 + t2) % 2017; 
		t1 = t2; 
		t2 = now; 
	}
	
	cout << now << endl ;
}